app/uk/gov/hmrc/transitmovements/repositories/MovementsRepository.scala

```diff
@@ -306,16 +330,34 @@ class MovementsRepositoryImpl @Inject() (
       case Success(obs) =>
         obs
           .toFuture()
-          .map(
-            response => Right(response.toVector)
-          )
+          .map {
+            response =>
+              val (start, end) = indices(pageNumber, itemCount, response.length)
```

Comment:
I didn't realise this when we looked last week, but why aren't we getting the database to do this bit? Given the P4 database has about 120k entries (https://grafana.tools.production.tax.service.gov.uk/d/transits-movements-trader-at-departure/transits-movements-trader-at-departure?orgId=1&refresh=15m&viewPanel=60), I don't think it would be too far fetched to assume that there could be something on the order of few thousand.

[Can we make the database do it using $slice?](https://www.mongodb.com/docs/manual/reference/operator/aggregation/slice/#mongodb-expression-exp.-slice) I'd imagine that would save a lot of memory and time in the long run, as the database would be optimised to do this.